src/pages/Index.js

```diff
@@ -66,53 +123,161 @@ modalWithImage.setEventListeners();
 /*             UserInfo                  */
 /*                                       */
 
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({ name: res.name, description: res.about });
+    userInfo.setAvatar(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 const userInfo = new UserInfo({
   profileTitle: ".profile__title",
   profileDescription: ".profile__description",
+  avatarSelector: ".profile__avatar",
 });
 
 /*                                       */
-/*             card.js                   */
+/*           ProfileEditModal            */
 /*                                       */
 
-function renderCard(cardData) {
-  const addCard = new Card(cardData, cardSelector, handleImageClick);
-  return addCard.getView();
+const editModal = new ModalWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+
+editModal.setEventListeners();
+
+function handleProfileEditSubmit(formData) {
+  editModal.renderLoading(true);
+  api
+    .updateProfileInfo({ name: formData.title, about: formData.description })
+    .then(() => {
+      userInfo.setUserInfo({
+        name: formData.title,
+        description: formData.description,
+      });
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.renderLoading(false);
+    });
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    settings,
-    renderer: renderCard,
-  },
+/*                                       */
+/*              avatar                   */
+/*                                       */
+
+function handleAvatarSubmit(data) {
+  editAvatarModal.renderLoading(true);
+  api
+    .updateAvatar({ avatar: data.link })
+    .then((res) => {
+      userInfo.setAvatar(res.avatar);
+      editAvatarModal.close();
+    })
+    .catch((err) => {
+      console.error("Failed to update user avatar:", err);
+    })
+    .finally(() => {
+      editAvatarModal.renderLoading(false);
+    });
+}
 
-  ".cards__list"
+const editAvatarModal = new ModalWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
 );
+editAvatarModal.setEventListeners();
 
-cardSection.renderItems();
+profileAvatarContainer.addEventListener("click", () => {
+  editAvatarModal.open();
+});
 
 /*                                       */
-/*             Event Handlers            */
+/*          LIKE & DISLIKE               */
 /*                                       */
 
+function handleLikeClick(card) {
+  if (card.getIsLiked()) {
```

Comment:
Needs correction: if the card is currently liked you should remove the like otherwise - remove (now everything looks confusing, maybe you need to change the naming in your Api class)